pallets/pools/src/lib.rs

```diff
@@ -628,6 +684,42 @@ pub mod pallet {
 				None => None,
 			};
 
+			for (tranche, tranche_input) in tranches.tranches.iter().zip(&tranche_inputs) {
+				let token_name: BoundedVec<u8, T::MaxTokenNameLength> = tranche_input
+					.clone()
+					.metadata
+					.token_name
+					.try_into()
+					.map_err(|_| Error::<T>::BadMetadata)?;
```

Comment:
I would be more granular with this error. 

```suggestion
					.map_err(|_| Error::<T>::TrancheTokenNameTooLong)?;
```